docs/proposals/08-shoot-additional-container-runtimes.md

```diff
@@ -0,0 +1,159 @@
+        3. Add a generator that will override the default containerd.service unit. It will create a new unit in: /etc/systemd/system/containerd.service unit that will start the containerd service with a configuration file path: /etc/containerd/config.toml.
+        
+    3. os-suse-jeos - __TBD__ 
+    
+3. Docker-monitor daemon and rotate log should be replaced with equivalent conatinerd services. __TBD__ 
+4. In order to install additional runtimes on the worker machine we must provide the installation scripts to the machine file system and run them.
+   The installation script should download and install the runtime binaries in each OS.
+   Instructions for Gardener to copy files to the machine filesystem and execute them is done via the cloud config secret.
+   
+   1. Each OS extension should implement a generator that will copy a generic packages installation script file.
+        ```yaml
+          #cloud-config
+          write_files:
+            path: '/var/lib/packages/install.sh'
+            ...
+      ```
+      
+      The installation file will be executed with different parameters defining the relevant package that should be installed:
+        1. Package name
+        2. Package path url
+        3.  __TBD__ 
+        
+   2. Each Extension controller should reconcile the OperatingSystemConfig (Should be enhanced to contain additional runtime data __TBD__):
+        1. Create a service unit to run the runtime binaries package installation script with the relevant parameters (e.g name of runtime container. package url).
+           ```
+           [Unit]
+           Description=Download and install additional runtimes binaries
+           ...
+       
+           [Service]
+           ExecStart=/var/lib/packages/install.sh <Parameters>
+           ...
+           ```
+        2. Copy a container runtime configuration script file and create a unit to run it. For example:
+           ```yaml
+           #cloud-config
+           write_files:
+           path: '/var/lib/kata-containers/configure.sh'
+           ...
+           [Unit]
+           Description=Configure 
+           ...
+      
+           [Service]
+           ExecStart=/var/lib/kata-containers/configure.sh
+           ```
+           
+           The installation script will be responsible for configuring the config.toml plugins at /etc/containerd/config.toml to contain the relevant runtime plugins.
+                1. kata-containers -  __TBD__
+                2. gvisor - __TBD__
+  
+5. Installation of a container runtime in the cluster will be done by the extension controllers. Each runtime type will be represented as an Extension resource. For example:
+      ```yaml
+      apiVersion: core.gardener.cloud/v1beta1
+      kind: ControllerRegistration
+      metadata:
+        name: extenstion-kata-containers-runtime
+      spec:
+        resources:
+        - kind: Extension
```

Comment:
We should introduce a new extension kind. `Extension` is more for generic addons like the shoot DNS or cert service.
What about `kind: ContainerRuntime`?

Reply:
Thank you, it sounds good. The conciliation of this new extension should apply the runtime classes to the cluster and it is quite similar to other add-on (Like nginx ingress) do you agree?

Reply:
Yes, sounds great 👍